canonical-web-and-design/canonical.com
# Add dynamic navigation to careers page

## Done

- Dynamically generate the sidebar navigation on the canonical.com/careers pages from the external departments list on the Greenhouse Harvest API.
- Added generic template for careers pages without overview content.

## QA

- Check out this feature branch
- You will need to add a Harvest API key to your local `.env.local` file (create one if it does not exist) under the name HARVEST_API_KEY. If you need one, let me know and I'll send one to you.
- Run the site using the command `./run`
- View the site locally in your web browser at: http://0.0.0.0:8002/careers/all
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Ensure that the sidebar navigation contains all the fields that are currently in the live version
- Ensure that the vacancy counter in the sidebar is working properly and is suitably formatted
- Ensure that the pages with existing templates render properly (for example, all those which are currently on the live site), including those which are not department pages such as http://0.0.0.0:8002/careers/lifestyle
- Ensure that all relevant vacancies still show up properly
- Ensure that the default template is suitably formatted (at this time, I believe http://0.0.0.0:8002/careers/product-management is the only page that uses this template)
- Ensure that the error page for nonexistent departments (such as http://0.0.0.0:8002/careers/none) is suitable
- Ensure that the code is sensible, both semantically and in terms of formatting


## Issue / Card

Fixes #263
